marshal.go

```diff
@@ -106,7 +106,13 @@ func parsePort(s string) int {
 func (i *InstanceMetadata) UnmarshalJSON(b []byte) error {
 	i.Raw = b
 	// TODO(cq) could actually parse Raw here, and in a parallel UnmarshalXML as well.
```

Comment:
Update the comment to note the only outstanding proposal: unmarshal the XML. You took care of the first proposal here.